tools/wptserve/wptserve/response.py

```diff
@@ -243,7 +243,13 @@ def set_error(self, code, message=u""):
                         ("Content-Length", len(data))]
         self.content = data
         if code == 500:
-            self.logger.error(message)
+            if isinstance(message, str) and message:
+                first_line = message.splitlines()[0]
+            else:
+                first_line = "<no message given>"
+            self.logger.error("Exception loading %s: %s" % (self.request.url,
```

Comment:
I think it can be used for non-500 errors too (it's part of the response API so handlers can call it) and in practice there's a single exception handler in the request handling function that calls this. In any case out of scope for this patch.